### ISSUE TITLE
Localstack doesn't work with python 3.9

### ISSUE THREAD

**Issue by vringar** [mozilla/OpenWPM#858]
This issue just appeared for me a couple of days ago.
You might need to switch to docker based workflow/an older version of conda for the time being.
https://github.com/conda/conda/issues/9653

**Comment by boolean5** [comment]
I Just ran `repin.sh` with conda version 4.8.3 and I'm getting some errors about specifications being incompatible with each other and with the python installation in my environment. Here's the output: [repin_output.txt](https://github.com/mozilla/OpenWPM/files/5988304/repin_output.txt)

**Comment by vringar** [comment]
Paging @birdsarah as the conda expert in residence:
As you can see some of the conflicts are complete bogus:
```
Package pathlib2 conflicts for:
ipython -> pathlib2
pytest -> pathlib2[version='>=2.2.0']
pytest-cov -> pytest[version='>=4.6'] -> pathlib2[version='>=2.2.0']
pre-commit -> importlib_metadata -> pathlib2
```
The issue I have referenced conda v4.8.1 so we might need to go back another version, but that doesn't seem like a good way forward.

Any ideas? Do we say that #753 removes the need for solving from a partially pinned environment and so we just merge the evironment.yamls and then solve from there?

**Comment by vringar** [comment]
Looking through conda/conda#9653 again someone mentions `--update-deps` as a workaround however neither `conda env create` nor `conda env update` support this flag

**Comment by birdsarah** [comment]
The latest version of conda is 4.9.2. If you're not on that, start there.

**Comment by vringar** [comment]
I have the same issue on 4.9.2

**Comment by vringar** [comment]
It seems to be broken starting in 4.8.1 and has been broken ever since.

**Comment by birdsarah** [comment]
OK. I've run this locally. I don't see the problem. I'm not getting "frozen." When it tries to add the dev dependencies it hits conflicts. Here's the full output:

```
UnsatisfiableError: The following specifications were found to be incompatible with each other:

Output in format: Requested package -> Available versions

Package toml conflicts for:
black -> toml[version='>=0.10.1|>=0.9.4']
pytest-cov -> pytest[version='>=4.6'] -> toml
pytest -> toml
pre-commit -> toml

Package pytest conflicts for:
pytest-cov -> pytest[version='>=3.6|>=4.6']
pytest

Package ca-certificates conflicts for:
ipython -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
pip -> python -> ca-certificates
pytest-cov -> python -> ca-certificates
python=3.9 -> openssl[version='>=1.1.1i,<1.1.2a'] -> ca-certificates
pre-commit -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
pytest -> python[version='>=2.7,<2.8.0a0'] -> ca-certificates
codecov -> python -> ca-certificates

Package packaging conflicts for:
pytest -> packaging
pip -> wheel -> packaging[version='>=20.2']
pytest-cov -> pytest[version='>=4.6'] -> packaging

Package typing_extensions conflicts for:
pytest -> importlib-metadata[version='>=0.12'] -> typing_extensions[version='>=3.6.4']
black -> typing_extensions[version='>=3.7.4']

Package pathlib2 conflicts for:
pre-commit -> importlib_metadata -> pathlib2
pytest-cov -> pytest[version='>=4.6'] -> pathlib2[version='>=2.2.0']
ipython -> pathlib2
pytest -> pathlib2[version='>=2.2.0']

Package six conflicts for:
pytest -> more-itertools[version='>=4.0.0'] -> six[version='>=1.0.0,<2.0.0']
localstack==0.11.1.1 -> six
pre-commit -> virtualenv[version='>=20.0.8'] -> six[version='>=1.9.0,<2']
pytest -> six[version='>=1.10.0']
pre-commit -> six
ipython -> prompt-toolkit[version='>=2.0.0,!=3.0.0,!=3.0.1,<3.1.0'] -> six[version='>=1.9.0']
pytest-cov -> pytest[version='>=4.6'] -> six[version='>=1.10.0']

Package gdbm conflicts for:
ipython -> pypy3.7[version='>=7.3.3'] -> gdbm[version='>=1.18,<1.19.0a0']
pre-commit -> pypy3.6[version='>=7.3.3'] -> gdbm[version='>=1.18,<1.19.0a0']
pytest -> pypy3.7[version='>=7.3.3'] -> gdbm[version='>=1.18,<1.19.0a0']

Package wcwidth conflicts for:
ipython -> prompt-toolkit[version='>=2.0.0,!=3.0.0,!=3.0.1,<3.1.0'] -> wcwidth
pytest-cov -> pytest[version='>=4.6'] -> wcwidth
pytest -> wcwidth

Package pip conflicts for:
localstack==0.11.1.1 -> python[version='>=3.8,<3.9.0a0'] -> pip
black -> python[version='>=3.6'] -> pip
codecov -> python -> pip
pytest -> python[version='>=3.8,<3.9.0a0'] -> pip
pytest-cov -> python -> pip
python=3.9 -> pip
pre-commit -> python[version='>=3.6,<3.7.0a0'] -> pip
ipython -> python[version='>=3.7,<3.8.0a0'] -> pip

Package tzdata conflicts for:
pytest -> python[version='>=3.9,<3.10.0a0'] -> tzdata
ipython -> python[version='>=3.9,<3.10.0a0'] -> tzdata
codecov -> python -> tzdata
pre-commit -> python[version='>=3.9,<3.10.0a0'] -> tzdata
pip -> python[version='>=3.6'] -> tzdata
black -> python[version='>=3.6'] -> tzdata
pytest-cov -> python -> tzdata
python=3.9 -> tzdata

Package importlib-metadata conflicts for:
pytest -> importlib_metadata[version='>=0.12'] -> importlib-metadata[version='>=1.1.3,<1.1.4.0a0|>=1.5.0,<1.5.1.0a0|>=1.5.2,<1.5.3.0a0|>=1.6.0,<1.6.1.0a0|>=1.6.1,<1.6.2.0a0|>=1.7.0,<1.7.1.0a0|>=2.0.0,<2.0.1.0a0|>=3.0.0,<3.0.1.0a0|>=3.1.0,<3.1.1.0a0|>=3.1.1,<3.1.2.0a0|>=3.3.0,<3.3.1.0a0|>=3.4.0,<3.4.1.0a0']
pytest-cov -> pytest[version='>=4.6'] -> importlib-metadata[version='>=0.12']
pre-commit -> importlib_metadata -> importlib-metadata[version='>=0.12,<2|>=0.12,<3|>=0.12|>=1.1.3,<1.1.4.0a0|>=1.5.0,<1.5.1.0a0|>=1.5.2,<1.5.3.0a0|>=1.6.0,<1.6.1.0a0|>=1.6.1,<1.6.2.0a0|>=1.7.0,<1.7.1.0a0|>=2.0.0,<2.0.1.0a0|>=3.0.0,<3.0.1.0a0|>=3.1.0,<3.1.1.0a0|>=3.1.1,<3.1.2.0a0|>=3.3.0,<3.3.1.0a0|>=3.4.0,<3.4.1.0a0']
pytest -> importlib-metadata[version='>=0.12']

Package pypy3.7 conflicts for:
ipython -> python[version='>=3.7,<3.8.0a0'] -> pypy3.7[version='7.3.*|7.3.3.*']
ipython -> pypy3.7[version='>=7.3.3']

Package certifi conflicts for:
codecov -> requests[version='>=2.7.9'] -> certifi[version='>=2016.09|>=2016.9.26|>=2017.4.17']
pytest -> setuptools[version='>=40.0'] -> certifi[version='>=2016.09|>=2016.9.26']
pip -> setuptools -> certifi[version='>=2016.09|>=2016.9.26']
pytest-cov -> setuptools -> certifi[version='>=2016.09|>=2016.9.26']
ipython -> setuptools[version='>=18.5'] -> certifi[version='>=2016.09|>=2016.9.26']

Package setuptools conflicts for:
ipython -> setuptools[version='>=18.5']
ipython -> pygments -> setuptools

Package wheel conflicts for:
pip -> wheel
python=3.9 -> pip -> wheel

Package bzip2 conflicts for:
ipython -> python[version='>=3.7,<3.8.0a0'] -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
localstack==0.11.1.1 -> python[version='>=3.8,<3.9.0a0'] -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
codecov -> python -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
pre-commit -> pypy3.6[version='>=7.3.3'] -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
pip -> python[version='>=3.6'] -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
black -> python[version='>=3.6'] -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
pytest -> python[version='>=3.8,<3.9.0a0'] -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']
pytest-cov -> python -> bzip2[version='>=1.0.6,<2.0a0|>=1.0.8,<2.0a0']

Package attrs conflicts for:
pytest -> attrs[version='>=17.2.0|>=17.4.0,<=19.1.0|>=17.4.0|>=19.2.0']
pytest-cov -> pytest[version='>=4.6'] -> attrs[version='>=17.2.0|>=17.4.0,<=19.1.0|>=17.4.0|>=19.2.0']
black -> attrs[version='>=17.4.0|>=18.1.0']

Package typing conflicts for:
pre-commit -> importlib_resources -> typing
black -> mypy_extensions[version='>=0.4.3'] -> typing[version='>=3.5.3|>=3.7.4|>=3.6.2']

Package python_abi conflicts for:
ipython -> jedi[version='>=0.16,<1.0'] -> python_abi=2.7[build=*_cp27mu]
ipython -> python_abi[version='3.6|3.6.*|3.7|3.7.*|3.9.*|3.8.*',build='*_cp38|*_cp37m|*_cp39|*_pypy37_pp73|*_cp36m|*_pypy36_pp73']

Package pypy3.6 conflicts for:
ipython -> pypy3.6[version='>=7.3.1|>=7.3.3']
ipython -> jedi[version='>=0.16,<1.0'] -> pypy3.6[version='7.3.*|7.3.0.*|7.3.1.*|7.3.2.*|7.3.3.*|>=7.3.2']

Package scandir conflicts for:
ipython -> pathlib2 -> scandir
pytest -> pathlib2[version='>=2.2.0'] -> scandir

Package coverage conflicts for:
pytest-cov -> coverage[version='>=4.4|>=5.2.1']
codecov -> coverage

Package importlib_metadata conflicts for:
pytest -> importlib_metadata[version='>=0.12']
pytest -> pluggy[version='>=0.9'] -> importlib_metadata[version='>=0.9']

Package zipp conflicts for:
pytest -> importlib-metadata[version='>=0.12'] -> zipp[version='>=0.5']
pre-commit -> importlib_metadata -> zipp[version='>=0.3.2|>=0.5|>=0.4']
```

I'm more than happy to agree this isn't easy to parse though. Given that localstack's a trash fire I went hunting there first found `localstack==0.11.1.1 -> python[version='>=3.8,<3.9.0a0'] -> pip`. 

One of the things that not optimal - because we surpress output in repin.sh using `-q` it's not easy to see the version of python that was being installed. After pinning to `python<3.9` (as indicated by localstack) everything's fine with the following output from `conda list`

```
# packages in environment at /home/sbird/miniconda3/envs/openwpm:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                 conda_forge    conda-forge
_openmp_mutex             4.5                       1_gnu    conda-forge
abseil-cpp                20200923.3           h9c3ff4c_0    conda-forge
amazon-kclpy              2.0.1                    pypi_0    pypi
appdirs                   1.4.4              pyh9f0ad1d_0    conda-forge
argparse                  1.4.0                    pypi_0    pypi
arrow-cpp                 3.0.0           py38h2b3ae1c_3_cpu    conda-forge
attrs                     20.3.0             pyhd3deb0d_0    conda-forge
aws-c-cal                 0.4.5                h2ff795d_6    conda-forge
aws-c-common              0.4.67               h7f98852_0    conda-forge
aws-c-event-stream        0.2.6                h4285e0c_4    conda-forge
aws-c-io                  0.8.3                h3b39b8c_1    conda-forge
aws-checksums             0.1.11               h3b39b8c_1    conda-forge
aws-sam-translator        1.34.0                   pypi_0    pypi
aws-sdk-cpp               1.8.138              h9e0957a_1    conda-forge
aws-xray-sdk              2.6.0                    pypi_0    pypi
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.1                      py_0    conda-forge
beautifulsoup4            4.9.3              pyhb0f4dca_0    conda-forge
black                     20.8b1                     py_1    conda-forge
boto                      2.49.0                   pypi_0    pypi
boto3                     1.17.8             pyhd8ed1ab_0    conda-forge
botocore                  1.20.8             pyhd8ed1ab_0    conda-forge
brotli                    1.0.9                h9c3ff4c_4    conda-forge
brotlipy                  0.7.0           py38h497a2fe_1001    conda-forge
bzip2                     1.0.8                h7f98852_4    conda-forge
c-ares                    1.17.1               h36c2ea0_0    conda-forge
ca-certificates           2020.12.5            ha878542_0    conda-forge
certifi                   2020.12.5        py38h578d9bd_1    conda-forge
cffi                      1.14.5           py38ha65f79e_0    conda-forge
cfgv                      3.2.0                      py_0    conda-forge
cfn-lint                  0.45.0                   pypi_0    pypi
chardet                   4.0.0            py38h578d9bd_1    conda-forge
click                     7.1.2              pyh9f0ad1d_0    conda-forge
codecov                   2.1.11             pyhd3deb0d_0    conda-forge
coverage                  5.4              py38h497a2fe_0    conda-forge
crontab                   0.22.9                   pypi_0    pypi
cryptography              3.4.4            py38h3e25421_0    conda-forge
dataclasses               0.7                pyhb2cacf7_7    conda-forge
dataclasses-json          0.5.2                    pypi_0    pypi
decorator                 4.4.2                      py_0    conda-forge
dill                      0.3.3              pyhd8ed1ab_0    conda-forge
distlib                   0.3.1              pyh9f0ad1d_0    conda-forge
dnslib                    0.9.14             pyh9f0ad1d_0    conda-forge
dnspython                 2.0.0                      py_0    conda-forge
docker                    4.4.2                    pypi_0    pypi
docopt                    0.6.2                      py_1    conda-forge
domain-utils              0.7.1                    pypi_0    pypi
easyprocess               0.3              py38h32f6830_2    conda-forge
ecdsa                     0.14.1                   pypi_0    pypi
editdistance              0.5.3            py38h709712a_3    conda-forge
filelock                  3.0.12             pyh9f0ad1d_0    conda-forge
flask                     1.1.2                    pypi_0    pypi
flask-cors                3.0.10                   pypi_0    pypi
freetype                  2.10.4               h0708190_1    conda-forge
fsspec                    0.8.5              pyhd8ed1ab_0    conda-forge
future                    0.18.2                   pypi_0    pypi
geckodriver               0.29.0               h3146498_0    conda-forge
gettext                   0.19.8.1          h0b5b191_1005    conda-forge
gflags                    2.2.2             he1b5a44_1004    conda-forge
glog                      0.4.0                h49b9bf7_3    conda-forge
grpc-cpp                  1.35.0               h146f9af_0    conda-forge
icu                       68.1                 h58526e2_0    conda-forge
identify                  1.5.13             pyh44b312d_0    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
iniconfig                 1.1.1              pyh9f0ad1d_0    conda-forge
ipython                   7.20.0           py38h81c977d_2    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
itsdangerous              1.1.0                    pypi_0    pypi
jedi                      0.18.0           py38h578d9bd_2    conda-forge
jinja2                    2.11.3                   pypi_0    pypi
jmespath                  0.10.0             pyh9f0ad1d_0    conda-forge
jpeg                      9d                   h36c2ea0_0    conda-forge
jsondiff                  1.2.0                    pypi_0    pypi
jsonpatch                 1.28                     pypi_0    pypi
jsonpickle                2.0.0                    pypi_0    pypi
jsonpointer               2.0                      pypi_0    pypi
jsonschema                3.2.0                    pypi_0    pypi
junit-xml                 1.9                      pypi_0    pypi
krb5                      1.17.2               h926e7f8_0    conda-forge
lcms2                     2.12                 hddcbb42_0    conda-forge
ld_impl_linux-64          2.35.1               hea4e1c9_2    conda-forge
leveldb                   1.22                 h7cfaab3_1    conda-forge
libblas                   3.9.0                8_openblas    conda-forge
libcblas                  3.9.0                8_openblas    conda-forge
libcurl                   7.71.1               hcdd3856_8    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libevent                  2.1.10               hcdb4288_3    conda-forge
libffi                    3.3                  h58526e2_2    conda-forge
libgcc-ng                 9.3.0               h2828fa1_18    conda-forge
libgfortran-ng            9.3.0               hff62375_18    conda-forge
libgfortran5              9.3.0               hff62375_18    conda-forge
libgomp                   9.3.0               h2828fa1_18    conda-forge
libidn2                   2.3.0                h516909a_0    conda-forge
liblapack                 3.9.0                8_openblas    conda-forge
libnghttp2                1.43.0               h812cca2_0    conda-forge
libopenblas               0.3.12          pthreads_h4812303_1    conda-forge
libpng                    1.6.37               h21135ba_2    conda-forge
libprotobuf               3.14.0               h780b84a_0    conda-forge
libssh2                   1.9.0                hab1572f_5    conda-forge
libstdcxx-ng              9.3.0               h6de172a_18    conda-forge
libthrift                 0.13.0               h5aa387f_6    conda-forge
libtiff                   4.2.0                hdc55705_0    conda-forge
libunistring              0.9.10               h14c3975_0    conda-forge
libutf8proc               2.6.1                h7f98852_0    conda-forge
libuv                     1.40.0               h7f98852_0    conda-forge
libwebp-base              1.2.0                h7f98852_0    conda-forge
localstack                0.11.1.1         py38h32f6830_0    conda-forge
localstack-client         1.12               pyh44b312d_0    conda-forge
localstack-ext            0.11.14.3          pyh9f0ad1d_0    conda-forge
lz4-c                     1.9.3                h9c3ff4c_0    conda-forge
markupsafe                1.1.1                    pypi_0    pypi
marshmallow               3.10.0                   pypi_0    pypi
marshmallow-enum          1.5.1                    pypi_0    pypi
mock                      4.0.3                    pypi_0    pypi
more-itertools            8.7.0              pyhd8ed1ab_0    conda-forge
moto-ext                  1.3.15.15                pypi_0    pypi
multiprocess              0.70.11.1        py38h497a2fe_1    conda-forge
mypy_extensions           0.4.3            py38h578d9bd_3    conda-forge
ncurses                   6.2                  h58526e2_4    conda-forge
networkx                  2.5                      pypi_0    pypi
nodeenv                   1.5.0              pyh9f0ad1d_0    conda-forge
nodejs                    14.15.1              h25f6087_0    conda-forge
numpy                     1.20.1           py38h18fd61f_0    conda-forge
olefile                   0.46               pyh9f0ad1d_1    conda-forge
openssl                   1.1.1i               h7f98852_0    conda-forge
orc                       1.6.7                h7950760_0    conda-forge
packaging                 20.9               pyh44b312d_0    conda-forge
pandas                    1.2.2            py38h51da96c_0    conda-forge
parquet-cpp               1.5.1                         2    conda-forge
parso                     0.8.1              pyhd8ed1ab_0    conda-forge
pathspec                  0.8.1              pyhd3deb0d_0    conda-forge
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.1.0            py38ha0e1e83_2    conda-forge
pip                       21.0.1             pyhd8ed1ab_0    conda-forge
pluggy                    0.13.1           py38h578d9bd_4    conda-forge
plyvel                    1.3.0                    pypi_0    pypi
pre-commit                2.10.1           py38h578d9bd_0    conda-forge
prompt-toolkit            3.0.16             pyha770c72_0    conda-forge
psutil                    5.8.0            py38h497a2fe_1    conda-forge
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
py                        1.10.0             pyhd3deb0d_0    conda-forge
pyaes                     1.6.1                      py_0    conda-forge
pyarrow                   3.0.0           py38h842ea0c_3_cpu    conda-forge
pyasn1                    0.4.8                    pypi_0    pypi
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygments                  2.8.0              pyhd8ed1ab_0    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pyparsing                 2.4.7              pyh9f0ad1d_0    conda-forge
pyrsistent                0.17.3                   pypi_0    pypi
pysocks                   1.7.1            py38h578d9bd_3    conda-forge
pytest                    6.2.2            py38h578d9bd_0    conda-forge
pytest-cov                2.11.1             pyh44b312d_0    conda-forge
python                    3.8.6           hffdb5ce_5_cpython    conda-forge
python-dateutil           2.8.1                      py_0    conda-forge
python-jose               3.2.0                    pypi_0    pypi
python_abi                3.8                      1_cp38    conda-forge
pytz                      2021.1             pyhd8ed1ab_0    conda-forge
pyvirtualdisplay          1.3.2            py38h32f6830_1    conda-forge
pyyaml                    5.4.1            py38h497a2fe_0    conda-forge
re2                       2020.11.01           h58526e2_0    conda-forge
readline                  8.0                  he28a2e2_2    conda-forge
redis-py                  3.5.3              pyh9f0ad1d_0    conda-forge
regex                     2020.11.13       py38h497a2fe_1    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
requests-file             1.5.1                    pypi_0    pypi
responses                 0.12.1                   pypi_0    pypi
rsa                       4.7.1                    pypi_0    pypi
s2n                       0.10.26              h9b69904_0    conda-forge
s3fs                      0.4.0                      py_0    conda-forge
s3transfer                0.3.4              pyhd8ed1ab_0    conda-forge
selenium                  3.141.0         py38h497a2fe_1002    conda-forge
sentry-sdk                0.20.2             pyh44b312d_0    conda-forge
setuptools                49.6.0           py38h578d9bd_3    conda-forge
six                       1.15.0             pyh9f0ad1d_0    conda-forge
snappy                    1.1.8                he1b5a44_3    conda-forge
soupsieve                 2.0.1                      py_1    conda-forge
sqlite                    3.34.0               h74cdb3f_0    conda-forge
sshpubkeys                3.3.1                    pypi_0    pypi
stringcase                1.2.0                    pypi_0    pypi
subprocess32              3.5.4                    pypi_0    pypi
tabulate                  0.8.7              pyh9f0ad1d_0    conda-forge
tblib                     1.6.0                      py_0    conda-forge
tk                        8.6.10               h21135ba_1    conda-forge
tldextract                2.2.2                    pypi_0    pypi
toml                      0.10.2             pyhd8ed1ab_0    conda-forge
traitlets                 5.0.5                      py_0    conda-forge
typed-ast                 1.4.2            py38h497a2fe_0    conda-forge
typing-inspect            0.6.0                    pypi_0    pypi
typing_extensions         3.7.4.3                    py_0    conda-forge
urllib3                   1.26.3             pyhd8ed1ab_0    conda-forge
virtualenv                20.4.2           py38h578d9bd_0    conda-forge
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
websocket-client          0.57.0                   pypi_0    pypi
werkzeug                  1.0.1                    pypi_0    pypi
wget                      1.20.1               h22169c7_0    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
wrapt                     1.12.1                   pypi_0    pypi
xmltodict                 0.12.0                   pypi_0    pypi
xz                        5.2.5                h516909a_1    conda-forge
yaml                      0.2.5                h516909a_0    conda-forge
zipp                      3.4.0                    pypi_0    pypi
zlib                      1.2.11            h516909a_1010    conda-forge
zstd                      1.4.8                ha95c52a_1    conda-forge
```

I feel like there's an ongoing default assumption that conda's broken. I'm not saying it's perfect but it is a widely used piece of software.

**Comment by birdsarah** [comment]
Here's a branch with the updates: https://github.com/mozilla/OpenWPM/tree/birdsarah-test-partial-release

**Comment by birdsarah** [comment]
I did discover a separate issue in running this though. When I pinned python with `python<3.9` everything worked correctly but crucially the pinned python dependency didn't make it into the pinned `environment.yaml`. When I instead used `python==3.8.6` it did make it in (see this commit 5a0e025). That seems like a bug in `prune-environmeny.py` that should be fixed and/or noted in docs.

**Comment by vringar** [comment]
I don't think that conda is broken by default. I much prefer it to our old ways of managing dependencies and like the sleek and reliable way of managing dependencies that it offers. I'm sorry that I was too negative/harsh in the words I used.
This issue just stopped me while trying to finish up work I'm already late on.
Thank you so much for figuring this out.

**Comment by vringar** [comment]
In conclusion: the culprit is localstack.
The remediation is to pin python to <3.9 (which should be done to close this issue) and to merge the storage refactor

**Comment by vringar** [comment]
Closed by #859
